scene/resources/skeleton_modification_2d.cpp

```diff
@@ -0,0 +1,3433 @@
+}
+
+int SkeletonModification2D::get_execution_mode() const {
+	return execution_mode;
+}
+
+void SkeletonModification2D::set_editor_draw_gizmo(bool p_draw_gizmo) {
+	editor_draw_gizmo = p_draw_gizmo;
+#ifdef TOOLS_ENABLED
+	if (is_setup) {
+		stack->set_editor_gizmos_dirty(true);
+	}
+#endif // TOOLS_ENABLED
+}
+
+bool SkeletonModification2D::get_editor_draw_gizmo() const {
+	return editor_draw_gizmo;
+}
+
+void SkeletonModification2D::_bind_methods() {
+	BIND_VMETHOD(MethodInfo("execute", PropertyInfo(Variant::FLOAT, "delta")));
+	BIND_VMETHOD(MethodInfo("setup_modification", PropertyInfo(Variant::OBJECT, "modification_stack", PROPERTY_HINT_RESOURCE_TYPE, "SkeletonModificationStack2D")));
+	BIND_VMETHOD(MethodInfo("draw_editor_gizmo"));
+
+	ClassDB::bind_method(D_METHOD("set_enabled", "enabled"), &SkeletonModification2D::set_enabled);
+	ClassDB::bind_method(D_METHOD("get_enabled"), &SkeletonModification2D::get_enabled);
+	ClassDB::bind_method(D_METHOD("get_modification_stack"), &SkeletonModification2D::get_modification_stack);
+	ClassDB::bind_method(D_METHOD("set_is_setup", "is_setup"), &SkeletonModification2D::set_is_setup);
+	ClassDB::bind_method(D_METHOD("get_is_setup"), &SkeletonModification2D::get_is_setup);
+	ClassDB::bind_method(D_METHOD("set_execution_mode", "execution_mode"), &SkeletonModification2D::set_execution_mode);
+	ClassDB::bind_method(D_METHOD("get_execution_mode"), &SkeletonModification2D::get_execution_mode);
+	ClassDB::bind_method(D_METHOD("clamp_angle", "angle", "min", "max", "invert"), &SkeletonModification2D::clamp_angle);
+	ClassDB::bind_method(D_METHOD("set_editor_draw_gizmo", "draw_gizmo"), &SkeletonModification2D::set_editor_draw_gizmo);
+	ClassDB::bind_method(D_METHOD("get_editor_draw_gizmo"), &SkeletonModification2D::get_editor_draw_gizmo);
+
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "enabled"), "set_enabled", "get_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "execution_mode", PROPERTY_HINT_ENUM, "process, physics_process"), "set_execution_mode", "get_execution_mode");
+}
+
+SkeletonModification2D::SkeletonModification2D() {
+	stack = nullptr;
+	is_setup = false;
+}
+
+///////////////////////////////////////
+// LookAt
+///////////////////////////////////////
+
+bool SkeletonModification2DLookAt::_set(const StringName &p_path, const Variant &p_value) {
+	String path = p_path;
+
+	if (path.begins_with("enable_constraint")) {
+		set_enable_constraint(p_value);
+	} else if (path.begins_with("constraint_angle_min")) {
+		set_constraint_angle_min(Math::deg2rad(float(p_value)));
+	} else if (path.begins_with("constraint_angle_max")) {
+		set_constraint_angle_max(Math::deg2rad(float(p_value)));
+	} else if (path.begins_with("constraint_angle_invert")) {
+		set_constraint_angle_invert(p_value);
+	} else if (path.begins_with("constraint_in_localspace")) {
+		set_constraint_in_localspace(p_value);
+	} else if (path.begins_with("additional_rotation")) {
+		set_additional_rotation(Math::deg2rad(float(p_value)));
+	}
```

Comment:
Why use custom `_set` / `_get`? Godot provides methods and members bindings, for this kind of things.

It's not an issue convert that parameter either, because you can always do it inside the `set`/`get` functions.

Note: It's already done, and I don't want you to force change this part, though consider that so to use Godot standards.

Reply:
This is the Godot standards, using setter and getter functions with `_set` and `_get` for properties is used all throughout the Godot code base. This allows you to expose the functions to GDScript using properties, which I believe isn't possible using just `_set` and `_get`.

Reply:
Using `_set` and `_get` is only for implementing custom properties which can't use the standard bindings.

For all these properties, it looks like they could just use macros like this in `_bind_methods` instead:
`ADD_PROPERTY(PropertyInfo(Variant::BOOL, "enable_constraint"), "set_enable_constraint", "get_enable_constraint");`

It would also make things exposed to script more consistent. For instance in your implementation, if I'm not mistaken, calling `set_constraint_angle_min(angle)` from gdscript would take a rad value, while `constraint_angle_min = angle` is expecting a value in degrees.